packages/common/src/search/_internal/portalSearchGroupMembers.ts

```diff
@@ -0,0 +1,190 @@
+import { serializeQueryForPortal } from "../serializeQueryForPortal";
+import {
+  IHubSearchOptions,
+  IHubSearchResponse,
+  IHubSearchResult,
+  IQuery,
+} from "../types";
+import {
+  ISearchGroupUsersOptions,
+  IUser,
+  getUser,
+  searchGroupUsers,
+} from "@esri/arcgis-rest-portal";
+import { expandPredicate } from "./expandPredicate";
+import { getNextFunction } from "../utils";
+import HubError from "../../HubError";
+import { IHubRequestOptions } from "../../types";
+import { enrichUserSearchResult } from "../../users";
+import { failSafe } from "../../utils";
+import { pickProps, setProp } from "../../objects";
+
+/**
+ * Search for members of a group.
+ * The group is specified via `IQuery.properties.groupId`
+ *
+ * The backing API is very limited in what
+ * it returns so this method executes the search and then tries to fetch
+ * the user object directly. This is a bit slower but provides a more
+ * information. Even in this case, private users may not be returned, and
+ * so we have a default user that is returned in those cases.
+ *
+ * @param query
+ * @param options
+ * @returns
+ */
+export async function portalSearchGroupMembers(
+  query: IQuery,
+  options: IHubSearchOptions
+): Promise<IHubSearchResponse<IHubSearchResult>> {
+  if (!query.properties?.groupId) {
+    throw new HubError(
+      "portalSearchGroupMembers",
+      "Group Id required. Please pass as query.properties.groupId"
+    );
+  }
+
+  const groupId = query.properties.groupId;
+
+  // Expand the individual predicates in each filter
+  query.filters = query.filters.map((filter) => {
+    // only `name` and `memberType` are supported
+    const validPredicateKeys = ["name", "memberType"];
```

Comment:
I may end up extracting this predicate scrubbing so it can be used to limit predicates for the `user` search as well